core/drand_daemon.go

```diff
@@ -58,6 +58,19 @@ func NewDrandDaemon(c *Config) (*DrandDaemon, error) {
 		beaconProcesses: make(map[string]*BeaconProcess),
 	}
 
+	// Add callback to registera new handler for http server after finishing DKG successfully
+	c.dkgCallback = func(share *key.Share, group *key.Group) {
+		beaconID := group.ID
+		if beaconID == "" {
+			beaconID = common.DefaultBeaconID
+		}
+
+		bp, isPresent := drandDaemon.beaconProcesses[beaconID]
```

Comment:
Don't you need a lock for this callback ?

Reply:
Yes, you are right! Thanks!